src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -64,10 +69,80 @@ void parallel_reduce (const TeamMember& team,
 
    result = local_tmp;
   } else {
+    const ValueType initial(result);
     Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+    result += initial;
   }
 }
 
+/*
+ * Computes a reduction over a routine described by 'input' for entries 'scalarize(input)([begin,end))'
+ * The variable 'input' should be an operator s.t. 'input(k)' returns a Pack. The computed reduction is
+ * added to the value 'result'.
+ */
+template <bool Serialize, typename TeamMember, typename InputProvider, typename ValueType>
+static KOKKOS_INLINE_FUNCTION
+void view_reduction (const TeamMember& team,
+                     const int begin,
+                     const int end,
+                     const InputProvider& input,
+                     ValueType& result)
+{
+  using PackType = typename std::remove_reference<decltype(input(0))>::type;
+  constexpr int vector_size = PackType::n;
+
+  // Perform a packed reduction over scalar indices
+  if (true) { // !Serialize) {
```

Comment:
Since I took the time to implement a serialized version of all functions called here (`parallel_reduce` and `reduce_sum`), if there are no objections I vote we do the packed reduction for both serial and parallel computations, instead of only for parallel (and a single loop over indices for serial, see commented out else-statement below). It makes the entire function simpler.

If this is ok, I'll remove the `if(true)` here and commented out else-statement below. If you don't agree, I'm happy to keep both routines with the `if(!Serialized)` check.

Reply:
I think it would be best to move the `Serialized` check to just the middle of the three blocks. If not `Serialized` then reduce at the pack level followed by reduction of the pack, like Luca does in Hommexx; this is more efficient than reducing every pack. If `Serialized`, use what you have. This way, we get maximum efficiency but reduce the code repetition to just the easy middle packs.

Reply:
I agree with Andrew. No need to reduce each pack if we are going for performance (i.e., Serialized=false). If not serialized, perform a packed reduction, then reduce the result pack at the end.

Reply:
For this I needed to add a definition for `reduction_identity<PackType>::sum()` to `ekat_kokkos_types.hpp`. Let me know if my implementation is acceptable.

Reply:
I don't think you pushed that mod. I don't see it in the list of touched files...